src/Aspire.Hosting.Yarp/YarpServiceExtensions.cs

```diff
@@ -0,0 +1,69 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Yarp;
+using Microsoft.Extensions.Hosting;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding YARP resources to the application model.
+/// </summary>
+public static class YarpServiceExtensions
+{
+    /// <summary>
+    /// Adds a YARP container to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<YarpResource> AddYarp(
+        this IDistributedApplicationBuilder builder,
+        [ResourceName] string name)
+    {
+        var resource = new YarpResource(name);
+
+        var yarpBuilder = builder.AddResource(resource)
+                      .WithHttpEndpoint(targetPort: YarpContainerImageTags.Port)
+                      .WithImage(YarpContainerImageTags.Image)
+                      .WithImageRegistry(YarpContainerImageTags.Registry)
+                      .WithEnvironment("ASPNETCORE_ENVIRONMENT", builder.Environment.EnvironmentName)
+                      .WithOtlpExporter();
+
+        if (builder.Environment.IsDevelopment())
+        {
+            // YARP will not trust the cert used by aspire otlp endpoint when running locally
+            yarpBuilder.WithEnvironment("YARP_UNSAFE_OLTP_CERT_ACCEPT_ANY_SERVER_CERTIFICATE", "true");
+        }
+
+        // Map the configuration file
+        yarpBuilder.ApplicationBuilder.Eventing.Subscribe<AfterEndpointsAllocatedEvent>((e, ct) =>
+        {
+            if (yarpBuilder.Resource.ConfigFilePath != null)
+            {
+                yarpBuilder.WithBindMount(yarpBuilder.Resource.ConfigFilePath, YarpContainerImageTags.ConfigFilePath, isReadOnly: true);
```

Comment:
@danegsta we should be using container files for this in dev right?

Reply:
Yeah, WithContainerFiles would make sense here. Currently the config file would have to be read and the contents passed to WithContainerFiles (but I'm working on getting support for existing files added for 9.3).